internal/disk/customizations.go

```diff
@@ -0,0 +1,104 @@
+package disk
+
+import (
+	"io"
+	"math/rand"
+
+	"github.com/google/uuid"
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/distro"
+)
+
+const (
+	sectorSize            = 512
+	BIOSBootPartitionGUID = "21686148-6449-6E6F-744E-656564454649"
+	BIOSBootPartitionUUID = "FAC7F1FB-3E8D-4137-A512-961DE09A5549"
+
+	FilesystemDataGUID = "0FC63DAF-8483-4772-8E79-3D69D8477DE4"
+	FilesystemDataUUID = "CB07C243-BC44-4717-853E-28852021225B"
+
+	EFISystemPartitionGUID = "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+	EFISystemPartitionUUID = "68B2905B-DF3E-4FB3-80FA-49D1E773AA33"
+	EFIFilesystemUUID      = "7B77-95E7"
+
+	RootPartitionUUID = "6264D520-3FB9-423F-8AB8-7A0A8E3D3562"
+)
+
+func CreatePartitionTable(
+	mountpoints []blueprint.FilesystemCustomization,
+	imageSize uint64,
+	archName string,
+	basePartitionTable PartitionTable,
+	rng *rand.Rand,
+) PartitionTable {
+
+	basePartitionTable.Size = imageSize
+	partitions := []Partition{}
+
+	if bootPartition := basePartitionTable.BootPartition(); bootPartition != nil {
+		// the boot partition UUID needs to be set since this
+		// needs to be randomly generated
+		bootPartition.Filesystem.UUID = uuid.Must(newRandomUUIDFromReader(rng)).String()
+	}
+
+	// start point for all of the arches is
+	// 2048 sectors.
+	var start uint64 = basePartitionTable.updatePartitionStartPointOffsets(2048)
```

Comment:
It would be great if you manage to handle the root partition inside of `updatePartitionStartPointOffsets()` to not split the logic into two functions, especially when only one is method of the `PartitionTable` type.